lib/rtcpeerconnection/edge.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+var EventTarget = require('../util/eventtarget');
+var inherits = require('util').inherits;
+var util = require('../util');
+
+var match = navigator.userAgent.match(/Edge\/(\d+).(\d+)$/);
+var PeerConnection = require('rtcpeerconnection-shim')(window, match[2]);
+
+function EdgeRTCPeerConnection(configuration) {
+  if (!(this instanceof EdgeRTCPeerConnection)) {
+    return new EdgeRTCPeerConnection(configuration);
+  }
+
+  EventTarget.call(this);
+
+  var peerConnection = new PeerConnection(configuration);
+
+  Object.defineProperties(this, {
+    _peerConnection: {
+      value: peerConnection
+    },
+    iceConnectionState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceConnectionState;
+      }
+    },
+    iceGatheringState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceGatheringState;
+      }
+    },
+    signalingState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.signalingState;
+      }
+    }
+  });
+
+  util.proxyProperties(PeerConnection.prototype, this, peerConnection);
+}
+
+inherits(EdgeRTCPeerConnection, EventTarget);
+
+// otalk/RTCPeerConnection shim throws InvalidAccessError
+// when sender is not part of getSenders or has been previously removed
```

Comment:
@syerrapragada the behavior you describe is also captured by Chromium [Issue 860853](https://bugs.chromium.org/p/chromium/issues/detail?id=860853). You may consider opening a PR to otalk/rtcpeerconnection-shim proposing the change.